File: StateBuilderJava/Examples/TrafficLight/src/main/java/com/stateforge/statemachine/examples/trafficlight/Light.java
package com.stateforge.statemachine.examples.trafficlight;

public class Light {

    private long redDuration = 1000;
    private long yellowDuration = 1000;
    private long greenDuration = 1000;
    private long maxDuration = 3500;
    
    public void turnOnGreen()
    {
        System.out.println("TurnOnGreen");
    }

    public void turnOffGreen()
    {
        System.out.println("TurnOffGreen");
    }

    public void turnOnYellow()
    {
        System.out.println("TurnOnYellow");
    }

    public void turnOffYellow()
    {
        System.out.println("TurnOffYellow");
    }
    public void turnOnRed()
    {
        System.out.println("TurnOnRed");
    }

    public void turnOffRed()
    {
        System.out.println("TurnOffRed");
    }

    public void setRedDuration(long redDuration) {
        this.redDuration = redDuration;
    }

    public long getRedDuration() {
        return redDuration;
    }

    public void setYellowDuration(long yellowDuration) {
        this.yellowDuration = yellowDuration;
    }

    public long getYellowDuration() {
        return yellowDuration;
    }

    public void setGreenDuration(long greenDuration) {
        this.greenDuration = greenDuration;
    }

    public long getGreenDuration() {
        return greenDuration;
    }

    public void setMaxDuration(long maxDuration) {
        this.maxDuration = maxDuration;
    }

    public long getMaxDuration() {
        return maxDuration;
    }
    
}

File: StateBuilderCpp/examples/qt/TrafficLight/Light.cpp


#include <QsLog.h>
#include <QsLogDest.h>

#include <Light.h>

Light::Light() :
  m_lRedDuration(1000),
  m_lYellowDuration(1000),
  m_lGreenDuration(1000),
  m_lMaxDuration(3500)
{
}

Light::~Light()
{
}

void Light::TurnOnRed()
{
  QLOG_INFO();
}

void Light::TurnOffRed()
{
  QLOG_INFO();
}

void Light::TurnOnYellow()
{
  QLOG_INFO();
}

void Light::TurnOffYellow()
{
  QLOG_INFO();
}

void Light::TurnOnGreen()
{
  QLOG_INFO();
}

void Light::TurnOffGreen()
{
  QLOG_INFO();
}

#include "Light.moc"
